src/Tracer.ts

```diff
@@ -0,0 +1,293 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Valentin Genev <valentin.genev@modusbox.com>
+
+ --------------
+ ******/
+
+'use strict'
+
+import { Trace, TraceContext } from "./Trace";
+import { EventLoggingServiceClient, SimpleLoggingServiceClient } from "./transport/EventLoggingServiceClient";
+import { EventType, NullEventAction, AuditEventAction, LogEventAction, TraceEventAction, EventMessage, EventTraceMetadata, IEventTrace, IEventMessage, EventAction, EventStateMetadata, EventMetadata, LogResponseStatus } from "./model/EventMessage";
+
+const Config = require('./lib/config');
+
+/**
+ * Logger Options sets the interface for the different logger options, which might be passed to the logger for different actions
+ */
+interface LoggerOptions {
+  action?: EventAction,
+  state?: EventStateMetadata
+}
+
+/**
+ * Implements the methods for user to work with tracing and logging. Sends all messages to the EventLoggingServiceClient.
+ * It provides null implementation of EventPreProcessor and EventPostProcessor.
+ * It can be extended to implement some of these methods.
+ */
+class Tracer extends Trace {
```

Comment:
I suggest that we engineer these two distinct objects separately for Trace vs Span, and use a composite relationship with Span being part of the Trace. Let me know what you think about this?

I feel that there is much confusion as to what is a Trace vs a Span in the above implementation.